src/OrbitBase/include/OrbitBase/TemporaryFile.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TEMPORARY_FILE_H_
+#define ORBIT_BASE_TEMPORARY_FILE_H_
+
+#include "OrbitBase/File.h"
+
+namespace orbit_base {
+class TemporaryFile final {
+ public:
+  TemporaryFile() { Create(); }
```

Comment:
Why not simply implement the ctor?

Also: The Create() method has a bunch of error conditions. It might make sense to add a method to query whether a TemporaryFile was successfully created. A static factory might be an alternative:

`static std::unique_ptr<TemporaryFile> Create() {...}`